### PR TITLE
modify preserve bounds logic

### LINKED ISSUE #316: [Bug]: regridder storing NaN for lat bounds
### What happened?

It seems like (in some instances) the horizontal regridder is saving latitude bounds with `NaN` values.

### What did you expect to happen?

Since we are passing in a grid with valid `lat_bnds`, these should be saved to the regridded dataset. 

### Minimal Complete Verifiable Example

```python
# imports
from xcdat.regridder import grid
import xcdat as xc
import numpy as np

# example file
fn = '/p/user_pub/climate_work/pochedley1/cesm2le/tas/tas_Amon_CESM2_historical-ssp370_1251-005_gr_185001-210012.nc'

# create target grid
ngrid = grid.create_grid(np.arange(-88.75, 90, 2.5), np.arange(1.25, 360., 2.5))

# open file
ds = xc.open_dataset(fn)
dsr = ds.regridder.horizontal('tas', ngrid, tool='xesmf', method='bilinear', periodic=True)

# notice that spatial averaging produces NaNs
print(dsr.spatial.average('tas').tas.values)                                                                                          
```

> [nan nan nan ... nan nan nan]

```
# this likely results because bounds_lat has NaN values
print(dsr.bounds_lat)
```

> <xarray.DataArray 'bounds_lat' (lat: 72, bound: 2)>
> array([[nan, nan],
>        [nan, nan],
>        [nan, nan],
>        [nan, nan],
>        ...

### Relevant log output

_No response_

### Anything else we need to know?

A separate `lat_bnds` variable is saved in the regridded dataset (it was not in the original dataset) with the correct bounds, but the `ds.lat.bounds` attribute is `bounds_lat`. 

This may be as simple as using that attribute to choose the name of the new bounds (if it exists) when saving the new, regridded dataset. 

For some reason this doesn't seem to affect longitude (in this example). `ds.bounds_lon` has the right shape and does not have NaN values. 

### Environment

main branch

### PR DESCRIPTION
**Pull request by pochedls**
## Description
This resolves an issue in which the bounds are populated by NaN values after regridding. The issue was related to functionality to preserve the source grid bounds in the target grid. Sometimes this functionality would result in multiple bounds (with different names) being saved for a single spatial axis (though I'm not sure why the values were NaN). 

- Closes #316
## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: xcdat/regridder/base.py
Comment (pochedls) [inline xcdat/regridder/base.py:26]: Did I rename things correctly here? I think we are passing in the target grid here (even though it is initially being used to "source" some bounds information).

File: xcdat/regridder/base.py
Comment (pochedls) [inline xcdat/regridder/base.py:59]: I skip copying the `X` and `Y` bounds information, because I think they should always be populated in the logic above from the target grid. Is this an okay assumption? Regardless, I think the source grid horizontal bounds are no longer applicable?

File: tests/test_regrid.py
Comment (pochedls) [inline tests/test_regrid.py:645]: If my logic makes sense, then we should not expect horizontal grid bounds to be preserved.

**Review by pochedls — COMMENTED** [review]
This is my quick assessment / fix, but this deserves review. I can also try to test it more broadly before we merge.

File: xcdat/regridder/base.py
Comment (pochedls) [reply xcdat/regridder/base.py:59]: I guess one question here: can you regrid just `X` or `Y` (e.g., if I have [`time`, `lat`] datasets)? I think in this scenario you could still skip copying the source dataset `X` and `Y` bounds.

**Comment by pochedls** [comment]
FYI – I did loop over a bunch of `tas` files and CDAT/xcdat agreed in every instance.

**Comment by tomvothecoder** [comment]
@jasonb5 @pochedls, do you both think this will be merged today or tomorrow?
Just checking if it can make 0.3.2, which I am targeting by tomorrow.

**Review by jasonb5 — CHANGES_REQUESTED** [review]

File: tests/test_regrid.py
Comment (pochedls) [reply tests/test_regrid.py:645]: For some reason it says that @jasonb5 requested changes (I think because of this comment), but I think @jasonb5 made these changes in his last commit.

**Comment by pochedls** [comment]
I think this can be merged upon approval from @jasonb5 and @tomvothecoder

File: tests/test_regrid.py
Comment (tomvothecoder) [reply tests/test_regrid.py:666]: Jason marked requested changes in the review before making the commit, so it will still show up as requested changes until he marks approved.

Example of how to request changes from a review:

image

**Review by tomvothecoder — APPROVED** [review]
Looks good to me. Thanks for this PR and @jasonb5 for reviewing.

You can go ahead and merge.

**Review by jasonb5 — APPROVED** [review]

File: tests/test_regrid.py
Comment (jasonb5) [reply tests/test_regrid.py:666]: @tomvothecoder @pochedls Yea I forgot to approve the changes after I made them. I'll get this merged in.

**Pull request merged**
